test/document_stores/test_weaviate.py

```diff
@@ -14,162 +21,207 @@ def get_uuid():
+    @pytest.mark.integration
+    def test_delete_labels(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_delete_labels_by_id(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_delete_labels_by_filter(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_delete_labels_by_filter_id(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_get_label_count(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_write_labels_duplicate(self):
+        pass
+
+    @pytest.mark.skip(reason="Weaviate does not support labels")
+    @pytest.mark.integration
+    def test_write_get_all_labels(self):
+        pass
+
+    @pytest.mark.integration
+    def test_ne_filters(self, ds, documents):
+        """
+        Weaviate doesn't include documents if the field is missing,
+        so we customize this test
+        """
+        ds.write_documents(documents)
+
+        result = ds.get_all_documents(filters={"year": {"$ne": "2020"}})
+        assert len(result) == 3
+
+    @pytest.mark.integration
+    def test_nin_filters(self, ds, documents):
+        """
+        Weaviate doesn't include documents if the field is missing,
+        so we customize this test
+        """
+        ds.write_documents(documents)
+
+        result = ds.get_all_documents(filters={"year": {"$nin": ["2020", "2021", "n.a."]}})
+        assert len(result) == 0
+
+    @pytest.mark.integration
+    def test_delete_index(self, ds, documents):
+        """Contrary to other Document Stores, this doesn't raise if the index is empty"""
+        ds.write_documents(documents, index="custom_index")
+        assert ds.get_document_count(index="custom_index") == len(documents)
+        ds.delete_index(index="custom_index")
+        assert ds.get_document_count(index="custom_index") == 0
+
+    @pytest.mark.integration
+    def test_query_by_embedding(self, ds, documents):
+        ds.write_documents(documents)
+
+        docs = ds.query_by_embedding(np.random.rand(embedding_dim).astype(np.float32))
+        assert len(docs) == 9
+
+        docs = ds.query_by_embedding(np.random.rand(embedding_dim).astype(np.float32), top_k=1)
+        assert len(docs) == 1
+
+        docs = ds.query_by_embedding(np.random.rand(embedding_dim).astype(np.float32), filters={"name": ["name_1"]})
+        assert len(docs) == 3
+
+    @pytest.mark.integration
+    def test_query(self, ds, documents):
+        ds.write_documents(documents)
+
+        query_text = "Foo"
+        docs = ds.query(query_text)
+        assert len(docs) == 3
+
+        # BM25 retrieval WITH filters is not yet supported as of Weaviate v1.14.1
```

Comment:
Version number in this comment should be upgraded.

Reply:
Let's do this in a different PR, opened an issue for that https://github.com/deepset-ai/haystack/issues/3553